cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -387,6 +397,26 @@ instance ToJSON TxIn where
 instance ToJSONKey TxIn where
   toJSONKey = toJSONKeyText renderTxIn
 
+instance FromJSON TxIn where
+  parseJSON = withText "TxIn" $ \txinStr -> runParsecParser parseTxIn txinStr
+
+parseTxId :: Parsec.Parser TxId
+parseTxId = do
+  str <- Parsec.many1 Parsec.hexDigit Parsec.<?> "transaction id (hexadecimal)"
```

Comment:
So it takes two hex digits to make a byte.  I'm curious what happens if we supply an odd number of digits?

Reply:
`parseTxId` fails but:
```
>let testParser = many1 hexDigit <?> "transaction id (hexadecimal)"
>parse testParser "" "a"

<interactive>:25:21: warning: [-Wtype-defaults]
    • Defaulting the following constraint to type ‘[ghc-prim-0.6.1:GHC.Types.Char]’
        Data.String.IsString s0 arising from the literal ‘"a"’
    • In the third argument of ‘parse’, namely ‘"a"’
      In the expression: parse testParser "" "a"
      In an equation for ‘it’: it = parse testParser "" "a"

<interactive>:25:21: warning: [-Wtype-defaults]
    • Defaulting the following constraint to type ‘[ghc-prim-0.6.1:GHC.Types.Char]’
        Data.String.IsString s0 arising from the literal ‘"a"’
    • In the third argument of ‘parse’, namely ‘"a"’
      In the expression: parse testParser "" "a"
      In an equation for ‘it’: it = parse testParser "" "a"
Right "a"